lib/shoulda/matchers/active_record/have_db_column_matcher.rb

```diff
@@ -137,6 +138,13 @@ def description
 
         protected
 
+        OPTIONS = %i(precision limit default null scale primary)
+
+        def validate_options(opts)
+          invalid_options = opts.keys.map(&:to_sym) - OPTIONS
+          raise ArgumentError, "Unknown option(s) '#{invalid_options}'" unless invalid_options.empty?
```

Comment:
Just a little style thing: we don't tend to use modifiers or `unless` — would you mind formatting this as:

``` ruby
if invalid_options.any?
  raise ArgumentError, "Unknown option(s) '#{invalid_options}'"
end
```

(although see note below about possibly reformatting the message)